src/sparsebase/preprocess/preprocess.h

```diff
@@ -826,6 +826,120 @@ class Degrees_DegreeDistribution
   void Register();
 };
 
+//! An abstract class representing partitioning algorithms.
+/*!
+ * Class that generalizes partitioning algorithms. It defines the API used for
+ * partitioning as well as the return type of partitioning (IDType*).
+ * @tparam IDType  the data type of row and column numbers (vertex IDs in the
+ * case of graphs)
+ */
+template <typename IDType>
+class PartitionPreprocessType : public FunctionMatcherMixin<IDType *> {
+public:
+  PartitionPreprocessType();
+
+  //! Performs a partition operation using the default parameters
+  /*!
+   * @returns An IDType array where the i-th index contains the ID for the partitioning i belongs to
+   */
+  IDType* Partition(format::Format * format, std::vector<context::Context*> contexts, bool convert_input);
```

Comment:
Actually, i was thinking about implementing the class that will apply a partitioning, and maybe it would be useful for partitioning to also return the number of parts. What do you think?

Reply:
Great idea. But lets open a separate issue for this, this PR waited long enough :)